android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncMixedPerformanceTestDefinitions.kt

```diff
@@ -0,0 +1,312 @@
+                    // stream being open indicates that the doc configs are now set as stale.
+                    // Check every 10ms so we're not doing too much work on this thread, and
+                    // don't log anything, so as not to pollute the test results with logging
+                    // overhead.
+                    ctx.testNetworkMonitor.connectedState = true
+                    var counter = 0
+                    while (!ctx.testDataSynchronizer.areAllStreamsOpen()) {
+                        Thread.sleep(10)
+
+                        // if this hangs longer than 30 seconds, throw an error
+                        counter += 1
+                        if (counter > 3000) {
+                            testHarness.logMessage("stream never opened after reconnect")
+                            error("stream never opened after reconnect")
+                        }
+                    }
+
+                    // Do the sync pass that will perform the stale document fetch
+                    val syncPassSucceeded = ctx.testDataSynchronizer.doSyncPass()
+
+                    // Perform syncPass() and halt the test if the pass fails
+                    if (!syncPassSucceeded) {
+                        error("sync pass failed")
+                    }
+                },
+                afterEach = { ctx, numDocs, _ ->
+                    // Verify that the test did indeed synchronize the updates locally
+                    SyncPerformanceTestUtils.assertLocalAndRemoteDBCount(ctx, numDocs)
+                }
+            )
+        }
+
+        /*
+         * Before: Perform remote insert of numDoc / 2 documents
+         *         Perform a local insert of numDoc / 2 documents
+         *         Configure sync(), perform sync pass
+         *         Update numChangeEvents / 2 documents remotely
+         *         Update numChangeEvents / 2 documents locally
+         *              Where numConflicts docs are updates on the same documents
+         * Test: Perform sync pass
+         * After: Ensure that the sync pass worked properly
+         */
+        fun testSyncPass(testHarness: SyncPerformanceIntTestsHarness, runId: ObjectId) {
+            // Run doTestSyncPass() for all changeEvent Percentages found in SyncPerfTestUtils
+            SyncPerformanceTestUtils.getChangeEventPercentages().forEach { changeEventPercentage ->
+                SyncPerformanceTestUtils.getConflictPercentages().forEach { conflictPercentage ->
+                    doTestSyncPass(testHarness, runId, changeEventPercentage, conflictPercentage)
+                }
+            }
+        }
+
+        private fun doTestSyncPass(
+            testHarness: SyncPerformanceIntTestsHarness,
+            runId: ObjectId,
+            pctOfDocsWithChangeEvents: Double,
+            pctOfDocsWithConflicts: Double
+        ) {
+            val testName = "Mixed_SyncPass"
```

Comment:
[meta q] What is a "Mixed_SyncPass"? Directions unclear :)

Reply:
A mixed sync pass is just updating a bunch of documents remotely and updating a bunch of documents locally with a set percentage of conflicts that will occur and then running a sync pass. Let me know if I should clarify this anywhere? I Have it in the comment block above

Reply:
These tests are basically named Load_TestName where load can be R2L, L2R, or Mixed and testName can be InitialSync, DisconnectReconnect, or SyncPass